KAS-4118: Add secretary data

sergiofenoll opened the pull request:
Adds property and relationships to the user model so that we can store the pertinent information.

sergiofenoll committed:
feat: add secretary relationships for user

ValenberghsSven commented on config/resources/users-domain.json:
```diff
@@ -64,6 +68,18 @@
+          "inverse": true,
+          "target": "decision-activity",
+          "cardinality": "many"
         }
```
should this be `prov:wasAssociatedWith` ?

ValenberghsSven commented on config/resources/users-domain.json:
```diff
@@ -34,6 +34,10 @@
         "created": {
           "type": "datetime",
           "predicate": "dct:created"
+        },
+        "is-current-secretary": {
+          "type": "boolean",
+          "predicate": "ext:isHuidigeSecretaris"
```
The ticket does not mention it, but I understood that the secretary user has a mandate just like the ministers.
Can we do the same thing as ministers and link a person to a user and then find the correct current mandate of that person?
I might be making this too complex, but I don't see the end goal yet so I want to prevent us from starting on the wrong foot.
This would be the same as setting  "isCurrentPrimeMinister" to true on the user  "Jan Jambon"

sergiofenoll replied:
Not sure, this might have been discusses while I was away. 
As far as I understand the secretary is not a minister. Although they might have a mandate inside the Flemish Government, I'm not sure if we're going to start tracking that inside Kaleidos. I can imagine that changing more often than the ministers, which would add an extra maintenance burden for us. Although it would be very cool complete reflection of the government's personnel as structured data inside Kaleidos/Themis!

> This would be the same as setting "isCurrentPrimeMinister" to true on the user "Jan Jambon"

Yes. Aside from modelling the secretaries as people with a mandate, I was also thinking about using a global "configuration" resource with such data, akin to the `EmailSettings` we currently have. Not sure if it's better, just another option.

tdn replied:
After today's discussion, that might indeed be the way to go (a new mandate). 

TBD before this is merged

ValenberghsSven commented:
Forgot to mention during review, but hte ticket also says that those relation are not to be propagated so yggdrasil changes are needed.

sergiofenoll committed:
chore: use correct predicate to link users to decision activities

sergiofenoll commented:
> Forgot to mention during review, but the ticket also says that those relations are not to be propagated so yggdrasil changes are needed.

I asked Tom about it and he said:
> Mja toen ik aan dat ticket aan het schrijven was bedacht ik me gewoon dat ik niet weet of het implicaties heeft als die informatie Kaleidos zou verlaten.

> Maar idd, eigenlijk is het wel publieke informatie wie in dat kabinet zit

Which is why I didn't make changes to Yggdrasil. We can double check on Monday to be sure.

kobemertens committed:
feat: change secretaris to mandatee instead of user

kobemertens committed:
feat: add secretaris roles

kobemertens committed:
chore: move secretaris to meeting instead of agenda

kobemertens committed:
feat: add secretaris mandate

kobemertens committed:
feat: add secretaris persons and mandates migrations

kobemertens committed:
fix: fix errors in secretaris migrations

SchrodingersCat00 commented:
These last three commits were added accidentally here, they are part of KAS-4118. Should I move them or do we just use this PR to build on?

kobemertens committed:
fix: add mandatees to correct government body

kobemertens committed:
chore: remove unused property

ValenberghsSven commented on config/migrations/20230717221117-add-secretaris-role.ttl:
```diff
@@ -0,0 +1,22 @@
+  skos:prefLabel "Gebruikersrollen"@nl .
+
+<http://themis.vlaanderen.be/id/bestuursfunctie/9d5ebfb9-3829-4b1f-a2a8-15033f7e2097> a org:Role ;
+    mu:uuid "9d5ebfb9-3829-4b1f-a2a8-15033f7e2097" ;
+    skos:inScheme thConceptScheme:b18acf1a-2a37-4b42-a549-b158d0065092 ;
+    skos:prefLabel "Vlaams Parlement" ;
+    skos:notation "Kaleidos_Parlement" ;
+    skos:definition "Secretaris" .
+
+<http://themis.vlaanderen.be/id/bestuursfunctie/cfa6ed74-bb6f-4d4c-b905-9a205be135d7> a org:Role ;
+    mu:uuid "cfa6ed74-bb6f-4d4c-b905-9a205be135d7" ;
+    skos:inScheme thConceptScheme:b18acf1a-2a37-4b42-a549-b158d0065092 ;
+    skos:prefLabel "Vlaams Parlement" ;
+    skos:notation "Kaleidos_Parlement" ;
+    skos:definition "Waarnemend Secretaris" .
```
met deze 2 erbij hebben we nu 3 `org:Role` (1 van vorige migratie) met dezelfde `prefLabel` en `notation`.
Voor kabinet hebben we duidelijk 2 verschillende rollen gemaakt beheerder en medewerker.
Willen we dat hier ook niet?
Ik heb nog niet gekeken naar de opvolgende PR's, niet zeker hoe dat verder gebruikt wordt.

sergiofenoll replied:
This should be `skos:prefLabel "Waarnemend Secretaris"`.

ValenberghsSven commented on config/migrations/20230719135250-add-secretaris-mandatees.ttl:
```diff
@@ -0,0 +1,62 @@
+
+<http://themis.vlaanderen.be/id/mandataris/f57a9d61-84ce-438e-b601-a905704ed50d> a mandaat:Mandataris ;
+	mandaat:isBestuurlijkeAliasVan	<http://themis.vlaanderen.be/id/persoon/3571cad5-15c8-4589-a0d8-2de1b6695b51> ;
+	mandaat:start	"2019-10-02T00:00:00Z"^^xsd:dateTime ;
```
kloppen deze starttijden? Moeten deze gelijk zijn met de huidige ministers (dus sinds aantreden brouns 18/05/2022)?

sergiofenoll replied:
Ik zou zeggen dat we ze zo laten en als we horen dat het anders moet kunnen we later extra mandaten toevoegen in het verleden. Voor huidige agenda's zal dat geen probleem zijn (we beschouwen de huidige mandatarissen als de meest recente, dan staat de link al juist) en dan hoeven we enkel te backfillen met secretarissen (die nergens aan gelinkt zijn).

ValenberghsSven replied:
voor mij ok. Denk niet dat die datums ergens tonen voor deze.

ValenberghsSven commented on config/migrations/20230718110645-add-secretaris-mandaat.ttl:
```diff
@@ -0,0 +1,14 @@
+    skos:prefLabel "secretaris" ;
+    org:role <http://themis.vlaanderen.be/id/bestuursfunctie/9d5ebfb9-3829-4b1f-a2a8-15033f7e2097> .
+
+<http://kanselarij.vo.data.gift/id/mandatarissen/14e0bcfa-b17d-48ea-b48c-effb862a3ae5> a mandaat:Mandaat ;
```
themis id?

sergiofenoll replied:
& Moet het niet `/mandaat/id` zijn ipv `mandatarissen`?

ValenberghsSven commented:
@Tom ging dit eens bekijken

ValenberghsSven committed:
Merge branch 'development' of github.com:kanselarij-vlaanderen/app-kaleidos into feature/KAS-4118-secretary-data

sergiofenoll commented on config/migrations/20230717221117-add-secretaris-role.ttl:
```diff
@@ -0,0 +1,22 @@
+
+thConceptScheme:b18acf1a-2a37-4b42-a549-b158d0065092 a skos:ConceptScheme ;
+  mu:uuid "b18acf1a-2a37-4b42-a549-b158d0065092" ;
+  skos:prefLabel "Gebruikersrollen"@nl .
```
"Gebruikersrollen" concept scheme doesn't make sense here. For other mandate roles (e.g. Minister-president) we use `http://data.vlaanderen.be/id/conceptscheme/BestuursfunctieCode`. See [this migration](https://github.com/kanselarij-vlaanderen/app-kaleidos/blob/development/config/migrations/20211113000000-themis-mandatarissen-kanselarij.ttl).

I think we will want to use that here as well.

sergiofenoll commented on config/migrations/20230717221117-add-secretaris-role.ttl:
```diff
@@ -0,0 +1,22 @@
+<http://themis.vlaanderen.be/id/bestuursfunctie/9d5ebfb9-3829-4b1f-a2a8-15033f7e2097> a org:Role ;
+    mu:uuid "9d5ebfb9-3829-4b1f-a2a8-15033f7e2097" ;
+    skos:inScheme thConceptScheme:b18acf1a-2a37-4b42-a549-b158d0065092 ;
+    skos:prefLabel "Vlaams Parlement" ;
+    skos:notation "Kaleidos_Parlement" ;
+    skos:definition "Secretaris" .
```
This should be `skos:prefLabel "Secretaris"`. Other triples can be left out.

sergiofenoll commented on config/migrations/20230718110645-add-secretaris-mandaat.ttl:
```diff
@@ -0,0 +1,14 @@
+
+<http://kanselarij.vo.data.gift/id/mandatarissen/64d19115-015e-42a8-9c73-6650f599ae8c> a mandaat:Mandaat ;
+    mu:uuid "64d19115-015e-42a8-9c73-6650f599ae8c" ;
+    skos:prefLabel "secretaris" ;
```
Mandates don't have a prefLabel.

sergiofenoll commented on config/migrations/20230718110645-add-secretaris-mandaat.ttl:
```diff
@@ -0,0 +1,14 @@
+
+<http://kanselarij.vo.data.gift/id/mandatarissen/14e0bcfa-b17d-48ea-b48c-effb862a3ae5> a mandaat:Mandaat ;
+    mu:uuid "14e0bcfa-b17d-48ea-b48c-effb862a3ae5" ;
+    skos:prefLabel "waarnemend secretaris" ;
```
Mandates don't have a prefLabel.

sergiofenoll commented on config/migrations/20230719135250-add-secretaris-mandatees.ttl:
```diff
@@ -0,0 +1,62 @@
+	mu:uuid	"c159b4ac-f033-48b6-a881-a13c3391ca6e" ;
+	dct:title	"Waarnemend Secretaris" .
+
+<http://themis.vlaanderen.be/id/bestuursorgaan/5fed907ee6670526694a0706> prov:hadMember <http://themis.vlaanderen.be/id/mandataris/f57a9d61-84ce-438e-b601-a905704ed50d> ,
```
Zijn de secretarissen deel van de regering? Not sure. @tdn heb jij een mening hierover?

ValenberghsSven replied:
Mijn idee hier was dat we net zoals ministers dan kunnen zoeken naar de juiste mandataris van de regeringsamenstelling van een bepaalde periode.
Op zich kunnen we dan bestaande logica opnieuw gebruiken in de frontend en moeten we gewoon andere rollen meegeven om op te filteren.

ValenberghsSven replied:
maar mogelijk is dit semantisch niet correct, niet zeker of dat gekoppeld mag worden idd.

sergiofenoll replied:
Het is op zich data die we niet releasen dus ik denk dat het niet zo'n groot probleem is als we dat nu zo instellen.

sergiofenoll committed:
fix: remove unnecessary concept scheme for roles

Mandate related roles (bestuursfuncties) are linked to the
data.vlaanderen.be concept scheme.

sergiofenoll committed:
fix: replace label triples with prefLabel for secretaris roles

sergiofenoll committed:
fix: use themis URIs and remove prefLabel for secretary mandate resources

ValenberghsSven commented on config/migrations/20230719135250-add-secretaris-mandatees.ttl:
```diff
@@ -0,0 +1,62 @@
+<http://themis.vlaanderen.be/id/mandataris/8c0f82ac-7424-4fde-be94-d0bdc9e30e1c> a mandaat:Mandataris ;
+	mandaat:isBestuurlijkeAliasVan	<http://themis.vlaanderen.be/id/persoon/1d4c8917-edc7-495f-a202-55511f459236> ;
+	mandaat:start	"2019-10-02T00:00:00Z"^^xsd:dateTime ;
+	org:holds	<http://kanselarij.vo.data.gift/id/mandatarissen/14e0bcfa-b17d-48ea-b48c-effb862a3ae5> ;
```
@sergiofenoll deze `org:holds` zijn nog de oude uris

ValenberghsSven commented:
Als je die holds aanpast kan je dan even ineens de timestamps aanpassen naar vandaag?
Anders staan die migraties zo uit volgorde.

sergiofenoll committed:
fix: replace URI to secretaris mandates

sergiofenoll committed:
chore: rename migrations to recent date

ValenberghsSven reviewed (approved):
This looks fine now
